include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -5914,26 +6010,29 @@ FEEvaluationAccess<dim, dim, Number, is_face, VectorizedArrayType>::
             divergence +=
               this->gradients_quad[(dim * d + d) * nqp + q_point] * inv_det;
         }
-      else if (this->cell_type <= internal::MatrixFreeFunctions::affine)
+      else
         {
-          // Affine cell
+          // General cell
           // Derivatives are reordered for faces. Need to take this into account
           const VectorizedArrayType inv_det =
             (is_face && dim == 2 && this->get_face_no() < 2) ?
-              -determinant(this->jacobian[0]) :
-              determinant(this->jacobian[0]);
+              -determinant(
+                this->jacobian[this->cell_type >
+                                   internal::MatrixFreeFunctions::affine ?
+                                 q_point :
+                                 0]) :
+              determinant(
+                this->jacobian[this->cell_type >
+                                   internal::MatrixFreeFunctions::affine ?
+                                 q_point :
+                                 0]);
```

Comment:
This looks a bit ugly due to the long formula. You could either introduce a short name for the index, say `q_index_geo`, or multiply by either `Number(-1.0)` or `Number(1.0)` depending on the two cases, so that `determinant(jacobian...)` can be stated once.